### PR TITLE
Add find_cells_intersecting_line

### LINKED ISSUE #4398: `find_cells_along_line` not working according to description
### Discussed in https://github.com/pyvista/pyvista/discussions/4390

<div type='discussions-op-text'>

<sup>Originally posted by **AsimAl-Sofi** May  6, 2023</sup>
Hi all
I would like to use the function **find_cells_along_line** to determine the cells along a line, but the function does not give me the right answer.
Here is my code example:
```py
out = pv.CylinderStructured(radius=(0,3),
theta_resolution=10,height=1,
z_resolution=2)
box=out.get_cell(5).get_face(3).cast_to_unstructured_grid().extract_surface()
box.compute_normals(inplace=True)
centers = out.get_cell(5).get_face(3).cast_to_unstructured_grid().cell_centers().points
arrows = box['Normals'][0]
pl = pv.Plotter()
pl.add_mesh(out, opacity=0.6, show_edges=True, line_width=1)
pl.add_points(arrows,color="g", point_size=18.0, render_points_as_spheres=True)
point_S = (0, -5, 2)
pl.add_points(np.array([point_S]),color="r", point_size=18.0, render_points_as_spheres=True)
line=pv.Line(point_S, arrows)
pl.add_mesh(line)
pl.show()
```

the function find_cells_along_line gives me the following answer:
```py
>>> out.find_cells_along_line(point_S, arrows,tolerance=0.0)
array([1, 0, 2, 7, 8, 9])
```
As you can see from the example, the line goes through cell 0, 8 and 9 but the function gives 1, 0, 2, 7, 8, 9.
Any suggestions how to tackle this problem?
Thanks</div>

### PR DESCRIPTION
**Pull request by MatthewFlamm**
### Overview

Closes #4398.  This PR first fixes the description of `find_cells_along_line` to clarify that it is finding cells whose bounds intersect the line and not the cells themselves.  The PR adds `find_cells_intersecting_line` to find cells that directly intersect the lines.

### Details

I updated the examples to be better.  The original example put the line in the (0, 0, 1) direction which should go through the vertex at the top of the sphere where many many cells converge together.  It is better to use the (1, 0, 0) direction where it should only intersect 1 cell.  The examples now highlight the difference between the methods where `find_cells_along_line` returns multiple cells, whereas `find_cells_intersecting_line` returns only 1 cell.

### CODE REVIEW

**Comment by MatthewFlamm** [comment]
cc @AsimAl-Sofi if you want to test this PR

File: pyvista/core/dataset.py [hunk 1]
```diff
@@ -2465,6 +2475,66 @@ def find_cells_along_line(
...
+        >>> mesh = pyvista.Sphere()
+        >>> mesh.find_cells_intersecting_line([0.0, 0, 0], [1.0, 0, 0])
+        array([896])
+
+        """
+        if pyvista.vtk_version_info < (9, 2, 0):
+            raise VTKVersionError("pyvista.PointSet requires VTK >= 9.2.0")
+
+        if np.array(pointa).size != 3:
+            raise TypeError("Point A must be a length three tuple of floats.")
```
Comment (MatthewFlamm) [inline pyvista/core/dataset.py:2526]: I followed the other method here, but I think it is probably better as `ValueError`.  If so, we should change both which would be a small breaking change.

Comment (akaszynski) [reply pyvista/core/dataset.py:2526]: Agree that `ValueError` would be correct here. It's out of scope of this PR, but let's do it in a different PR and have it merged for 0.40.0

**Review by akaszynski — APPROVED** [review]
Solid LGTM. Thanks @MatthewFlamm.

File: pyvista/core/dataset.py (hunk 1 above)
Comment (adeak) [reply pyvista/core/dataset.py:2526]: Well with the current wording ("length three tuple of floats") the type error is also correct. Which is just to say that we should probably reword it while we're at it.

Comment (akaszynski) [reply pyvista/core/dataset.py:2526]: We might consider reusing:

https://github.com/pyvista/pyvista/blob/461bb5ed5be8c9dda6e9ed765fab3224149d9c62/pyvista/utilities/errors.py#L414-L421

**Pull request merged**